File: TeacherEvaluation.py
import sys

n_students, target = map(int, input().split())
students = list(map(int, input().split()))
their_score = sum(students)
average = their_score/n_students

if average == 0 and their_score > 0:
    print("impossible")
    sys.exit()

if average == target:
    print(0)
    sys.exit()

for i in range(n_students+1, 5000):
    missing_score = target*i - their_score
    if 0 <= missing_score/(i - n_students) <= 100:
        print(i-n_students)
        break
else:
    print("impossible")

File: TeacherEvaluation.cpp
#include <iostream>
#include "bits/stdc++.h"
using namespace std;

int main() {
    int nStudents, target;
    cin >> nStudents >> target;
    vector<int> students;
    int temp, theirScore = 0;
    for (int i = 0; i < nStudents; ++i) {
        cin >> temp;
        theirScore += temp;
        students.push_back(temp);
    }

    if (theirScore == target*nStudents) {
        cout << 0 << endl;
        return 0;
    }

    for (int i = nStudents+1; i < 10000; ++i) {
        int missingScore = target*i - theirScore;
        double perStudent = (double)missingScore/(i-nStudents);
        if (perStudent <= 100 && perStudent >= 0) {
            cout << i-nStudents << endl;
            return 0;
        }
    }
    cout << "impossible" << endl;

    return 0;
}